### PR TITLE
#14437 allow passing in manual tax rate for CreatePOFromSOs process

### PR DESCRIPTION
**Pull request by adi-stefan**
https://github.com/metasfresh/metasfresh/issues/14437

### CODE REVIEW

File: backend/de.metas.acct.base/src/main/java/de/metas/acct/vatcode/VATCode.java [hunk 1]
```diff
@@ -34,23 +35,28 @@
 {
 	public static final VATCode NULL = new VATCode();
 
-	public static VATCode of(final String code)
+	public static VATCode of(final String code, final int vatCodeId)
```
Comment (teosarca) [inline backend/de.metas.acct.base/src/main/java/de/metas/acct/vatcode/VATCode.java:38]: boy scout rule-ish: could u pls remove `VATCode.NULL`

and change de.metas.acct.vatcode.impl.VATCodeDAO#findVATCode to return Optional<VATCode>

?

Comment (adi-stefan) [reply backend/de.metas.acct.base/src/main/java/de/metas/acct/vatcode/VATCode.java:38]: done.

**Review by teosarca — APPROVED** [review]
lgtm

**Comment by adi-stefan** [comment]
@Mergifyio backport master

**Pull request merged**